src/app/GitUI/Editor/FormFindInCommitFilesGitGrep.cs

```diff
@@ -124,6 +125,8 @@ private void SetSearchWatermarkLabelVisibility()
         lblSearchCommitGitGrepWatermark.Visible = cboFindInCommitFilesGitGrep.Visible && !cboFindInCommitFilesGitGrep.Focused && string.IsNullOrEmpty(cboFindInCommitFilesGitGrep.Text);
         if (lblSearchCommitGitGrepWatermark.Visible)
         {
+            int nonTextHeight = (cboFindInCommitFilesGitGrep.ClientRectangle.Height - cboFindInCommitFilesGitGrep.Font.Height) / 2;
+            lblSearchCommitGitGrepWatermark.Top = cboFindInCommitFilesGitGrep.Top + nonTextHeight;
```

Comment:
Any reason not to make the watermark the same size as the "owner" control?

We have other watermark controls, are we doing the same in those too?

Reply:
Owner control's size includes borders and the label would cover it. Setting transparent BackColor doesn't help, the "owner" control's border is simply not drawn where the label covers it even if it's transparent.

Tried some experiments
![image](https://github.com/user-attachments/assets/631b5819-3101-4a63-aa47-7b6fc73e8058)

FileStatusList looks at other control's bottom to determine position https://github.com/gitextensions/gitextensions/blob/master/src/app/GitUI/UserControls/FileStatusList.cs#L451-L453
and here https://github.com/gitextensions/gitextensions/blob/master/src/app/GitUI/UserControls/FileStatusList.cs#L472-L475
This PR's approach of placing the label vertically in the center of "owner" control works well and looks consistently between different scaling factors.

Reply:
RussKie meant controls having something like `EditNetSpell.WatermarkText` I guess. I could not find other such UserControls in a quick search.
FileStatusList with its hand-made dock layout is a rather unlucky example.

> This PR's approach of placing the label vertically in the center of "owner" control works well and looks consistently between different scaling factors.

Yes, but not perfectly. Alignment of the text baseline is a wide-spread issue in GE.

![image](https://github.com/user-attachments/assets/f8f49b78-48a9-47d2-bb17-687cb045fd6c)

Reply:
The reason I asked - we have several watermarks in various controls, I just want to make sure we have a consistent implementation. (Perhaps, there's even an opportunity to unify/generalise the implementations?)

Reply:
I am going to remove the watermark commit and address it separately in another PR. I think there is a way to handle these drop-down watermarks in the same way as `EditNetSpell.WatermarkText`